gcovr/tests/test_gcovr.py

```diff
@@ -380,7 +401,22 @@ def test_build(
         except Exception as e:  # pragma: no cover
             whole_diff_output += "  " + str(e) + "\n"
             if update_reference:
```

Comment:
I feel like the code for update-reference, generate-reference, and find-reference-files should each have their own function since they are closely related with each other, but distract from the main flow of the `test_build()` function.

Reply:
I'll refactor this.